pyproject.toml

```diff
@@ -214,6 +215,9 @@ ignore = [
 "notebooks/**/*.ipynb" = ["T201", "S101", "N811", "N802", "N803"]
 "src/y0/algorithm/separation/sigma_separation.py" = ["RUF002", "RUF003"]
 "src/y0/algorithm/separation/__init__.py" = ["RUF002", "RUF003"]
+"tests/**/*.py" = ["S101"]
+"tests/test_hierarchical.py" = ["N803", "N802", "N806"]
```

Comment:
I'm going to remove these ignores before making a review - these linting rules are to make sure that we have good variable names that follow a convention. Please carefully read through the feedback and try and pick better variable names